include/envoy/decompressor/decompressor.h

```diff
@@ -16,9 +16,10 @@ class Decompressor {
    * Decompresses data from one buffer into another buffer.
    * @param input_buffer supplies the buffer with compressed data.
    * @param output_buffer supplies the buffer to output decompressed data.
+   * @return the status code from decompression. A negative integer indicates failure.
```

Comment:
I'd suggest either returning a richer error status of some sort, either an enum described in this interface (ie not Z_OK et al) or bool+string for an expanded error message. Or you can just return a bool.

Reply:
I like the idea of mapping an enum to the library return codes.

Reply:
@derekargueta  are you going to make a commit to do that?